lektor/markdown.py

```diff
@@ -17,7 +18,10 @@ def link(self, link, title, text):
         if self.record is not None:
             url = url_parse(link)
             if not url.scheme:
-                link = self.record.url_to("!" + link, base_url=get_ctx().base_url)
+                # PRIMARY_ALT is used because images aren't generated for each alt
+                link = self.record.url_to(
+                    "!" + link, alt=PRIMARY_ALT, base_url=get_ctx().base_url
+                )
```

Comment:
I haven't tested this, and I am not that familiar with the alternative feature, but won't this also force all links to pages (that don't include a scheme) to the PRIMARY_ALT as well?  I'm pretty sure that would not be desired behavior.

I think what you really want to do here is remove the "!" prefixing from `link`.  This will (or should) cause `url_to` to resolve the path given by `link` to a db record (e.g. either a `Page` or an `Attachment`) and then ask that record for its URL (via its `.url_path` method).  The record will (or should) know the correct URL path to itself.

I've never understood why that `"!"` prefix is inserted here.  It seems like if the author had a use case where they wanted to prevent resolution through the db, they could add the `!` prefix themselves.  Even if there isn't a good reason for it — and there may be — it may be too late to change the behavior now.
To spitball: maybe we could introduce a new prefix which would force resolution through the db.  Or introduce a new _model option_ for the `markdown` type which when set would disable this automatic `!` prefixing.